presentation/src/main/java/com/anytypeio/anytype/presentation/spaces/SpaceListViewModel.kt

```diff
@@ -58,24 +93,106 @@ class SpaceListViewModel(
         initialValue = ViewState.Init
     )
 
+    fun onDeleteSpaceClicked(view: SpaceListItemView) {
+        warning.value = Warning.DeleteSpace(SpaceId(view.space.targetSpaceId!!))
+    }
+
+    fun onLeaveSpaceClicked(view: SpaceListItemView) {
+        warning.value = Warning.LeaveSpace(SpaceId(view.space.targetSpaceId!!))
+    }
+
+    fun onCancelJoinSpaceClicked(view: SpaceListItemView) {
+        warning.value = Warning.CancelSpaceJoinRequest(SpaceId(view.space.targetSpaceId!!))
+    }
+
+    fun onCancelJoinRequestAccepted(space: SpaceId) {
+        viewModelScope.launch {
+            Timber.d("Cancelling")
+            cancelJoinSpaceRequest.async(space).fold(
+                onSuccess = {
+                    Timber.d("Space deleted successfully")
+                },
+                onFailure = {
+                    Timber.e(it, "Error while deleting space")
+                }
+            )
+        }
+    }
+
+    fun onDeleteSpaceAccepted(space: SpaceId) {
+        viewModelScope.launch {
+            warning.update { Warning.None }
+            Timber.d("Deleting")
+            deleteSpace.async(space).fold(
+                onSuccess = {
+                    Timber.d("Space deleted successfully")
+                },
+                onFailure = {
+                    Timber.e(it, "Error while deleting space")
+                }
+            )
+        }
+    }
+
+    fun onLeaveSpaceAccepted(space: SpaceId) {
+        viewModelScope.launch {
+            warning.update { Warning.None }
+            Timber.d("Leaving")
+            deleteSpace.async(space).fold(
```

Comment:
Is it correct to use deleteSpace also here?

Reply:
yes